### PR TITLE
Delete the chat messages of a room when the room is deleted

### PR DESCRIPTION
**Pull request by danxuliu**

### CODE REVIEW

File: lib/AppInfo/Application.php [hunk 1]
```diff
@@ -110,4 +112,16 @@ protected function registerActivityHooks(EventDispatcherInterface $dispatcher) {
...
 	}
+
+	protected function registerChatHooks(EventDispatcherInterface $dispatcher) {
+		$listener = function(GenericEvent $event, $eventName) {
+			/** @var Room $room */
+			$room = $event->getSubject();
+
+			/** @var ChatManager $chatManager */
+			$chatManager = $this->getContainer()->query(ChatManager::class);
+			$chatManager->deleteMessages(strval($room->getId()));
```
Comment (nickvergessen) [inline lib/AppInfo/Application.php]: `(string) $room->getId()` please

File: lib/AppInfo/Application.php [hunk 2]
```diff
@@ -110,4 +112,16 @@ protected function registerActivityHooks(EventDispatcherInterface $dispatcher) {
...
+		};
+		$dispatcher->addListener(Room::class . '::preDeleteRoom', $listener);
```
Comment (nickvergessen) [inline lib/AppInfo/Application.php]: post please

File: lib/AppInfo/Application.php (hunk 1 above)
Comment (danxuliu) [reply lib/AppInfo/Application.php]: Done!

File: lib/AppInfo/Application.php (hunk 2 above)
Comment (danxuliu) [reply lib/AppInfo/Application.php]: Done!

**Pull request merged**